service/worker/worker.go

```diff
@@ -98,6 +98,9 @@ func (wm *workerManager) Start() {
 			// use default worker
 			wc.RegisterActivities(defaultWorker)
 		} else {
+			// TODO: This is to prevent issues during upgrade/downgrade. Remove in 1.24 release.
+			wc.RegisterActivities(defaultWorker)
```

Comment:
Do we need to register to `dedicatedWorker` as well? I guess no because all components we are switching are using defaultWorker. But plz double check.

Reply:
None of our existing components are using a `dedicatedWorker` (they all return `nil` from `DedicatedWorkflowWorkerOptions` but maybe we should still register there in case any community users have custom components?